core/src/main/scala/NodeResponse.scala

```diff
@@ -22,7 +22,7 @@ object NodeResponse {
         node            <- (j --\ "Node").as[String]
         address         <- (j --\ "Address").as[String]
         datacenter      <- (j --\ "Datacenter").as[String]
-        meta            <- (j --\ "Meta").as[Map[String, String]]
+        meta            <- (j --\ "Meta").as[Option[Map[String, String]]]
```

Comment:
Would you want to go so far as to have an `Option[NonEmptyMap[String, String]]` here?  Or is an empty map still possible?

Reply:
A good question. I don't know if it's actually possible to get this field to be empty JSON object inside the `Node` part of the response, as Consul always seems to set it to be `{"consul-network-segment": ""}` in my local instance (which also happens in my prod environment, except apparently at some point on Thursday last week).

That said, I was able to get `Meta` set to `{}` for a `Service` definition, so I'm going to assume that it could conceivably be empty.

The Consul API documentation does not specify either way, that I've been able to find.